x/emissions/keeper/actor_penalties.go

```diff
@@ -85,26 +105,39 @@ func ApplyLivenessPenaltyToActor(
 	nonceBlockHeight types.BlockHeight,
 	emaScore types.Score,
 ) (types.Score, error) {
+	if emaScore.BlockHeight == 0 {
+		ctx.Logger().Debug("no liveness penalty on new actor",
+			"nonce", nonceBlockHeight,
+			"score", emaScore,
+		)
+		return emaScore, nil
+	}
+
 	missedEpochs := missedEpochsFn(topic, emaScore.BlockHeight)
 	// No missed epochs == no penalty
 	if missedEpochs == 0 {
+		ctx.Logger().Debug("no liveness penalty on actor",
+			"nonce", nonceBlockHeight,
+			"score", emaScore,
+		)
 		return emaScore, nil
 	}
 
 	penalty, err := getAsymptoteFn(topic.Id)
 	if err != nil {
 		return types.Score{}, err
 	}
-	emaScore.BlockHeight = nonceBlockHeight
 
 	beforePenalty := emaScore
+	emaScore.BlockHeight = nonceBlockHeight
 	emaScore.Score, err = applyPenalty(topic, penalty, emaScore.Score, missedEpochs)
 	if err != nil {
 		return types.Score{}, err
 	}
 
 	ctx.Logger().Debug("apply liveness penalty on actor",
```

Comment:
is this logging format Ok ? Like actually showing the text but not the data?
It will compile but it may not show the data. Using fmt.Sprintf can build up the string ensuring that.

Reply:
Testing that right now

Reply:
This is definitely the preferred way, because it gives us the ability to destructure logs (either in json or logfmt) in Grafana and do lots of handy things with the extracted data.  If we're seeing the fields missing from the log output, that's a configuration issue.

Reply:
It is well structured!